Sources/Common/Communication/EventBus.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+
+/// Defines the contract for an event bus system.
+///
+/// This protocol outlines the core functionalities of an event bus, including posting events,
+/// adding observers, and managing observers. It is designed to support type-safe event handling
+/// and asynchronous execution
+public protocol EventBus: AutoMockable {
+    /// Posts an event to all registered observers.
+    ///
+    /// - Parameters:
+    ///   - event: The event to be posted.
+    /// - Returns: A Boolean indicating if the event was posted to any observers.
+    @discardableResult
+    func post(_ event: AnyEventRepresentable) async -> Bool
+    /// Adds an observer for a specific event type.
+    ///
+    /// - Parameters:
+    ///   - eventType: The event type to observe.
+    ///   - action: The action to execute when the event is observed.
+    func addObserver(_ eventType: String, action: @escaping (AnyEventRepresentable) -> Void) async
+    /// Removes all registered observers from the EventBus.
+    func removeAllObservers() async
+    /// Removes all observers for a specific event type.
+    ///
+    /// - Parameter eventType: The event type for which to remove observers.
+    func removeObserver(for eventType: String) async
+}
+
+/// A shared implementation of `EventBus` using an actor model for thread-safe operations.
+/// This actor manages the distribution of events to registered observers and uses
+/// `NotificationCenter` for event delivery. It ensures that event handling is thread-safe
+/// and observers are managed efficiently.
+// sourcery: InjectRegisterShared = "EventBus"
+// sourcery: InjectSingleton
+actor SharedEventBus: EventBus {
+    private var notificationCenter: NotificationCenter = .default
+    private var observers: [String: [NSObjectProtocol]] = [:]
+
+    deinit {
+        // Clean up by removing all observers when the EventBus is deinitialized.
+        // Capture a weak reference to self to avoid retain cycles
+        Task { [weak self] in
+            await self?.removeAllObservers()
+        }
```

Comment:
This bit challenged me a bit and I believe it won't do what you want it to do.
The problem here is that there is no guarantee that self won't be deallocated before the task is executed.
Hence, you will end up with `self == nil` before you get a chance to call `removeAllObservers`

I was thinking about not using weak; however, that's not great either as I have no idea what's the behavior of retaining `self` after `deinit` finish executing. All I can say is since `self` isn't expected to outlive `deinit` then nothing good will come out of retaining it. So we need an alternative approach here for the cleanup bit.
I will give it a bit of thoughts

Reply:
From the basic use-case testing, it did seem to deinit, but I agree there is no certainty that self will be present, but it won't crash but it wont do the job either. An alternative approach could be us not holding on to the reference of the notification center and letting components using it remove it before their `deinit`. But that brings an uncertainty that we won't know for sure if the observers have been removed/unsubscribed, because we will be at mercy of users of the events bus. 

I am thinking about an alternative too.

Reply:
> But that brings an uncertainty that we won't know for sure if the observers have been removed/unsubscribed

Yes, that will leave it for chance. I think extracting a tiny piece outside the actor to do the cleanup on its own deinit is a viable approach. We just need to make sure it doesn't include any sort of logic beyond the cleanup